code-challenges/CC5_string_methods/challenges-05.test.js

```diff
@@ -87,20 +93,25 @@ Write a function named splitFoods that uses split to produce the same output as
 You may also use other string or array methods.
 ------------------------------------------------------------------------------------------------ */
 
-const listFoods = (gruffaloCrumble) => {
-let result = [];
-
-//Loop through each index
-for (let i in gruffaloCrumble.ingredients.length) {
-  // With each index which is a string, turn it into an array. Split the array by spaces. .
-  let indexToArray = Array.of(gruffaloCrumble.ingredients[i]).split(' ');
-  // Identify the last item in the new individual arrays. Push it into the result.
-  result.push(indexToArray[indexToArray.length - 1]);
-  // Result is now an array of every ingredient.
+const splitFoods = (recipe) => {
+  let result = [];
+  let ingredients = recipe.ingredients; 
```

Comment:
This is a pass by reference. This means you will be modifying the original object data if you use any array methods that modify the array in place (which you do). You want to make a copy of the array so you don't make destructive changes. Once you fix this, later functions will behave more as you expected.